### ISSUE TITLE
xdmod ticket #29094 xdmid-setup

### ISSUE THREAD

**Issue by rob-baron** [CCI-MOC/xdmod-cntr#201]
Robert Bartlett Baron, reported over 1 year ago
Ok, created an xdmod instance, am attempting to set it up using xdmid-setup.

Here is the output from the script:

========================================================================

Database Setup

========================================================================

Please provide the information required to connect to your MySQL database

server.  A user will be created using the username and password you provide.

NOTE: The database password cannot include single quote characters (') or

double quote characters (").

DB Hostname or IP: [localhost] mariadb

DB Port: [3306] 

DB Username: [xdmod] 

DB Password: 

(confirm) DB Password: 

Please provide the password for the administrative account that will be

used to create the MySQL user and databases.

DB Admin Username: [root] 

DB Admin Password: 

(confirm) DB Admin Password: 

Creating User xdmod

Creating database `mod_shredder`.

Granting privileges on database `mod_shredder`.

Failed to create databases:

Command returned non-zero value '1': ERROR 1133 (28000) at line 1: Can't find any matching row in the user table

#0 /usr/share/xdmod/classes/CCR/DB/MySQLHelper.php(521): CCR\DB\MySQLHelper::staticExecuteCommand(Array)

#1 /usr/share/xdmod/classes/CCR/DB/MySQLHelper.php(476): CCR\DB\MySQLHelper::staticExecuteStatement('mariadb', '3306', 'root', 'pass', 'mod_shredder', 'GRANT ALL ON mo...')

#2 /usr/share/xdmod/classes/OpenXdmod/Shared/DatabaseHelper.php(116): CCR\DB\MySQLHelper::grantAllPrivilegesOnDatabase('mariadb', '3306', 'root', 'pass', 'mod_shredder', 'mariadb', 'xdmod', 'pass')

#3 /usr/share/xdmod/classes/OpenXdmod/Setup/DatabaseSetupItem.php(39): OpenXdmod\Shared\DatabaseHelper::createDatabases('root', 'pass', Array, Array, Object(OpenXdmod\Setup\Console))

#4 /usr/share/xdmod/classes/OpenXdmod/Setup/DatabaseSetup.php(109): OpenXdmod\Setup\DatabaseSetupItem->createDatabases('root', 'pass', Array, Array)

#5 /usr/share/xdmod/classes/OpenXdmod/Setup/Menu.php(69): OpenXdmod\Setup\DatabaseSetup->handle()

#6 /usr/bin/xdmod-setup(37): OpenXdmod\Setup\Menu->display()

#7 /usr/bin/xdmod-setup(22): main()

#8 {main}

Settings file not saved!

Press ENTER to continue. 

Ok, the script exists as it cannot create the databases, however, the first database it tries to create is mod_shredder (file: xdmod/classes/OpenXdmod/Setup/DatabaseSetup.php).

However, it is actually created:

MariaDB [(none)]> show databases;

+--------------------+

| Database           |

+--------------------+

| information_schema |

| mod_shredder       |

| mysql              |

| performance_schema |

| sys                |

+--------------------+

5 rows in set (0.00 sec)

And here are the users as existing after the script is run:

MariaDB [(none)]> select host, user, password from mysql.user;

+-----------+-------------+-------------------------------------------+

| Host      | User        | Password                                  |

+-----------+-------------+-------------------------------------------+

| localhost | mariadb.sys |                                           |

| localhost | root        | *196BDEDE2AE4F84CA44C47D54D78478C7E2BD7B7 |

| %         | root        | *196BDEDE2AE4F84CA44C47D54D78478C7E2BD7B7 |

| mariadb   | xdmod       | *196BDEDE2AE4F84CA44C47D54D78478C7E2BD7B7 |

+-----------+-------------+-------------------------------------------+

4 rows in set (0.01 sec)

Please advise as to how to proceed.

Jeffrey T. Palmer , said over 1 year ago
Ticket:  https://help.xdmod.org/support/tickets/29094

Hi Robert,

What version of MariaDB are you using?

Jeff

 

Jeffrey T. Palmer
Scientific Programmer
Center for Computational Research
701 Ellicott Street
Buffalo, NY 14203
716 881 7567
jtpalmer@buffalo.edu
buffalo.edu/ccr
Robert Bartlett Baron , said over 1 year ago
Server version: 10.6.5-MariaDB-1:10.6.5+maria~focal mariadb.org binary distribution

Jeffrey T. Palmer , said over 1 year ago
Ticket:  https://help.xdmod.org/support/tickets/29094

Hi Robert,

We don't yet support more recent versions of MariaDB.  Our current target is 5.5.68, the version in the CentOS 7 repositories.  Our next release, which should be out next month, will support 10.3 (i.e. CentOS 8, but due to the current status of CentOS we'll probably start targeting something else.)  I realize that we're quite behind the most recent releases, but that is where we are at.  If possible, please use the latest release of MariaDB 5.5.  We haven't tested with other versions so there may be other issues that we are not aware of.

The immediate issue I expect you are experiencing is due our use of implicit account creation (https://mariadb.com/kb/en/grant/#implicit-account-creation).  Check your sql_mode (SHOW VARIABLES LIKE '%sql_mode%';) and if you see NO_AUTO_CREATE_USER that is probably causing the issue.  You can try setting your sql_mode to '' (empty string) and try that.  That said, I'm not sure if there will be other compatibility issues.

Jeff

 

Jeffrey T. Palmer
Scientific Programmer
Center for Computational Research
701 Ellicott Street
Buffalo, NY 14203
716 881 7567
jtpalmer@buffalo.edu
buffalo.edu/ccr
Robert Bartlett Baron , said over 1 year ago
Ok, with mysql:5.5, I still get an error:

========================================================================
Database Setup
========================================================================

Please provide the information required to connect to your MySQL database
server.  A user will be created using the username and password you provide.

NOTE: The database password cannot include single quote characters (') or
double quote characters (").

DB Hostname or IP: [localhost] mariadb
DB Port: [3306] 
DB Username: [xdmod] 
DB Password: 
(confirm) DB Password: 

Please provide the password for the administrative account that will be
used to create the MySQL user and databases.

DB Admin Username: [root] 
DB Admin Password: 
(confirm) DB Admin Password: 
Creating User xdmod

Creating database `mod_shredder`.
Granting privileges on database `mod_shredder`.

Failed to create databases:

Command returned non-zero value '1': ERROR 1133 (42000) at line 1: Can't find any matching row in the user table

#0 /usr/share/xdmod/classes/CCR/DB/MySQLHelper.php(521): CCR\DB\MySQLHelper::staticExecuteCommand(Array)
#1 /usr/share/xdmod/classes/CCR/DB/MySQLHelper.php(476): CCR\DB\MySQLHelper::staticExecuteStatement('mariadb', '3306', 'root', 'pass', 'mod_shredder', 'GRANT ALL ON mo...')
#2 /usr/share/xdmod/classes/OpenXdmod/Shared/DatabaseHelper.php(116): CCR\DB\MySQLHelper::grantAllPrivilegesOnDatabase('mariadb', '3306', 'root', 'pass', 'mod_shredder', 'mariadb', 'xdmod', 'pass')
#3 /usr/share/xdmod/classes/OpenXdmod/Setup/DatabaseSetupItem.php(39): OpenXdmod\Shared\DatabaseHelper::createDatabases('root', 'pass', Array, Array, Object(OpenXdmod\Setup\Console))
#4 /usr/share/xdmod/classes/OpenXdmod/Setup/DatabaseSetup.php(109): OpenXdmod\Setup\DatabaseSetupItem->createDatabases('root', 'pass', Array, Array)
#5 /usr/share/xdmod/classes/OpenXdmod/Setup/Menu.php(69): OpenXdmod\Setup\DatabaseSetup->handle()
#6 /usr/bin/xdmod-setup(37): OpenXdmod\Setup\Menu->display()
#7 /usr/bin/xdmod-setup(22): main()
#8 {main}

Settings file not saved!

Press ENTER to continue. 

and likewise, the sql side:

[root@b60d7ae29374 /]# mysql -h mariadb --protocol=tcp -u root -p
Enter password: 
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 8
Server version: 5.5.64-MariaDB-1~trusty mariadb.org binary distribution

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [(none)]> show variables like '%sql_mode%' ;
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| sql_mode      |       |
+---------------+-------+
1 row in set (0.01 sec)

MariaDB [(none)]> show databases;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| mod_shredder       |
| mysql              |
| performance_schema |
+--------------------+
4 rows in set (0.00 sec)

So any ideas why mariadb:5.5 doesn't work?
Jeffrey T. Palmer , said over 1 year ago
Ticket:  https://help.xdmod.org/support/tickets/29094

Hi Robert,

I didn't realize you were using a non-localhost database.  The xdmod-setup script only supports creating a database on the localhost.  This probably isn't well documented enough, but you will need to create a database on the localhost and then move it to a different server and grant the privileges manually.  This is due to an unfortunate issue where I couldn't find an appropriate way of determining the correct grant statements that need to be run on the remote server.  I realize this is less than desirable, but it's another issue that we haven't yet been able to resolve.

Jeff

 

Jeffrey T. Palmer
Scientific Programmer
Center for Computational Research
701 Ellicott Street
Buffalo, NY 14203
716 881 7567
jtpalmer@buffalo.edu
buffalo.edu/ccr
Robert Bartlett Baron , said over 1 year ago
Jeffrey,

I found a way around this by running the following before xdmod-setup:

create user 'xdmod'@'mariadb' identified by 'pass';

create user 'xdmod'@'%' identified by 'pass';

create database mod_shredder;

create database mod_hpcdb;

create database moddb;

create database modw;

create database modw_aggregates;

create database modw_filters;

create database mod_logger;

grant all on mod_shredder.* to 'xdmod'@'mariadb';

grant all on mod_shredder.* to 'xdmod'@'%';

grant all on mod_hpcdb.* to 'xdmod'@'mariadb';

grant all on mod_hpcdb.* to 'xdmod'@'%';

grant all on moddb.* to 'xdmod'@'mariadb';

grant all on moddb.* to 'xdmod'@'%';

grant all on modw.* to 'xdmod'@'mariadb';

grant all on modw.* to 'xdmod'@'%';

grant all on modw_aggregates.* to 'xdmod'@'mariadb';

grant all on modw_aggregates.* to 'xdmod'@'%';

grant all on modw_filters.* to 'xdmod'@'mariadb';

grant all on modw_filters.* to 'xdmod'@'%';

grant all on mod_logger.* to 'xdmod'@'mariadb';

grant all on mod_logger.* to 'xdmod'@'%';

This is just a work-a-round.  I am planning on submitting a pull request to xdmod-setup when I find the time (once I get past this prototyping stage).

After this, I was able to run the xdmod-setup script, however, it could not overwrite the php.ini file as the file was read-only after the rpm install.

After some fiddling with the https.conf file, I was able to bring the interface up, as much as possible without any data.

We can close this issue.
Jeffrey T. Palmer , said over 1 year ago
Ticket:  https://help.xdmod.org/support/tickets/29094

Hi Robert,

Thanks for the update.  Let us know if you run into any other issues.

​Jeff

 

Jeffrey T. Palmer
Scientific Programmer
Center for Computational Research
701 Ellicott Street
Buffalo, NY 14203
716 881 7567
jtpalmer@buffalo.edu
buffalo.edu/ccr
This ticket has been Closed | over 1 year ago [Reply](https://help.xdmod.org/support/tickets/29094#reply-to-ticket)
Satisfaction Rating
We'd love to hear how your experience with our Support Team went. Whether or not your issue was resolved as you'd hoped, do you feel we did a good job in addressing your concerns? You can let us know by clicking on one of the three options below

[Great](https://help.xdmod.org/support/tickets/29094#)
[Just Okay](https://help.xdmod.org/support/tickets/29094#)
[Needs Improvement](https://help.xdmod.org/support/tickets/29094#)
Ticket details
Who do you need help from?

Type
Problem

Ticket Status
This ticket has been Closed

Priority
Low

Location of Script

Job ID Number

Application Used

Error Messages

[Home](https://help.xdmod.org/support/home) [Solutions](https://help.xdmod.org/support/solutions) [Tickets](https://help.xdmod.org/support/tickets)
